redfish/bootoption.go

```diff
@@ -0,0 +1,86 @@
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+package redfish
+
+import (
+	"encoding/json"
+
+	"github.com/stmcginnis/gofish/common"
+)
+
+// BootOption represents the properties of a bootable device available in the
+// system.
+type BootOption struct {
+	common.Entity
+
+	// ODataContext is the odata context.
+	ODataContext string `json:"@odata.context"`
+	// ODataType is the odata type.
+	ODataType string `json:"@odata.type"`
+	// Alias is the alias of this boot source if one exists.
+	Alias BootSourceOverrideTarget
+	// BootOptionEnabled is an indication of whether the boot option is
+	// enabled. If true , it is enabled. If false, the boot option that the
+	// boot order array on the computer system contains is skipped. In the
+	// UEFI context, this property shall influence the load option active
+	// flag for the boot option.
+	BootOptionEnabled bool
+	// BootOptionReference is the unique identifier seen in Boot.BootOrder.
+	BootOptionReference string
+	// DisplayName is the user-readable display name of the boot option
+	// that appears in the boot order list in the user interface.
+	DisplayName string
+	// UefiDevicePath is the UEFI device path to access this UEFI boot
+	// option.
+	UefiDevicePath string
+	// rawData holds the original serialized JSON so we can compare updates.
+	rawData []byte
```

Comment:
Looks like `rawData`, `etag`, and specialized `UnmarshalJSON` could all be removed since this is a simple object that does not have any of its own actions or specially serialized fields.

We could probably clean that up later though, and really no harm having it present other than a very slight extra overhead.